homeassistant/components/escea/config_flow.py

```diff
@@ -0,0 +1,44 @@
+"""Config flow for escea."""
+
+import asyncio
+from contextlib import suppress
+import logging
+
+from async_timeout import timeout
+
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import config_entry_flow
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from .const import DISPATCH_CONTROLLER_DISCOVERED, ESCEA, TIMEOUT_DISCOVERY
+from .discovery import async_start_discovery_service, async_stop_discovery_service
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def _async_has_devices(hass: HomeAssistant):
+
+    controller_ready = asyncio.Event()
+
+    @callback
+    def dispatch_discovered(_):
+        controller_ready.set()
+
+    async_dispatcher_connect(hass, DISPATCH_CONTROLLER_DISCOVERED, dispatch_discovered)
```

Comment:
This will leak because nothing is canceling this via the callable it returns

Reply:
Thanks - fixed